File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack = []
    for i in s:
    # 들어가기
        stack.append(i)
    # 나오기
        if len(stack) >= 2 and stack[-2] == '(' and i == ')':
            stack.pop()
            stack.pop()

    return False if len(stack) else True

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    let stack = []
    for(t of s){
        if(!stack || t == "("){
            stack.push(t)
        }else{
            if(stack[stack.length-1]=="("){
                stack.pop()
            }else{
                stack.push(t)
            }
        }
    }

    return stack.length==0?true:false;
}